src/com/jambit/onboarding2020/tbrpg/domain/Room/MobRoom.java

```diff
@@ -3,86 +3,185 @@
-      Enemy enemy = new Enemy(); // Name übergeben von Enemy
-      Random random = new Random();
+    @Override
+    public void printWelcomeMessage() {
+        System.out.println("Du stehst einem mächtigen Gegner gegenüber.");
+    }
 
-      System.out.println("Du bist gefangen und kommst nur raus, wenn du den Gegner tötest, oder 'quit' drückst");
-      System.out.println("Attack Damage Player: " + Player.getPlayerInstance().getHealthState());
-      System.out.println("Attack Damage Enemy: " + enemy.getHealthState());
+    @Override
+    public void enter() throws PlayerDeadException {
 
-      while (in.gameState()) {
-         System.out.println("Drücke 1) um anzugreifen oder 2) um zu verteidigen.");
 
-         try {
-            fight(in, enemy, random);
-         } catch (InvalidInputException e) {
-            System.out.println(e.getMessage());
-         }
+        GameInput in = new GameInput(new InputStreamReader(System.in));
+        Enemy enemy = new Enemy();
+        Weapon equippedWeapon = player.getEquippedWeapon();
 
-         evaluateFight(in, enemy);
-      }
-   }
+        System.out.println("Du bist gefangen und kommst nur raus, wenn du den Gegner tötest!" +
+                "\n--------------------------" +
+                "\nDein Gegner ist ein Alien... es sieht aus wie....." + enemy.getName() +
+                "\n" + enemy.getAsciiArt(enemy.getName()));
+        System.out.println("|Deine Lebenspunkte: " + player.getHealthState() + "|" +
+        "\n|Lebenspunkte Gegner: " + enemy.getHealthState() + "|" +
+        "\n|Equipped Weapon: " + player.getEquippedWeapon() + "|" +
+                "\n  _     _         __ _      _   _   _ \n" +
+                " | |___| |_ ___  / _(_)__ _| |_| |_| |\n" +
+                " | / -_)  _(_-< |  _| / _` | ' \\  _|_|\n" +
+                " |_\\___|\\__/__/ |_| |_\\__, |_||_\\__(_)\n" +
+                "                      |___/           ");
 
-   private void fight(GameInput in, Enemy enemy, Random random) throws InvalidInputException, PlayerDeadException {
-      int input = in.inputInteger();
-      ItemGenerator itemGenerator = new ItemGenerator();
 
-      if (input == 1) {
-         if (random.nextBoolean()) {
+        while (in.gameState()) {
+            System.out.println("Drücke 1) um anzugreifen oder 2) um das Pokemon zu streicheln");
+
+            try {
+                fight(in, enemy, equippedWeapon);
+            } catch (InvalidInputException e) {
+                System.out.println(e.getMessage());
+            }
+
+
+        }
+    }
+
+
+    private void fight(GameInput in, Enemy enemy, Weapon equippedWeapon) throws InvalidInputException, PlayerDeadException {
+        int input = in.inputInteger();
+        ItemGenerator itemGenerator = new ItemGenerator();
+
+        if (input == 1) {
             try {
-               Player.getPlayerInstance().attack(enemy);
+                player.attack(enemy, equippedWeapon);
```

Comment:
Man muss hier nicht equippedWeapon übergebn, da du einfach in der Methode ds Weapon abfragen kannst